### DISCUSSION TITLE
ESPHome light partition control

### DISCUSSION THREAD

**Discussion by asamicros** [zachowj/node-red-contrib-home-assistant-websocket#754] [Q&A] [answered]
Hi, I am trying to control a neopixel string that is partitioned into 2 lights.

I can control each light partition (12 pixels out of a 24 pixel string) from HA, and using a curl request.

But, using the call service node, all of the pixels are controlled, not just the partition.
[
    {
        "id": "a2d99facf5d519ff",
        "type": "debug",
        "z": "f3bbc3f8714fe9d3",
        "name": "debug 2",
        "active": true,
        "tosidebar": true,
        "console": false,
        "tostatus": false,
        "complete": "false",
        "statusVal": "",
        "statusType": "auto",
        "x": 900,
        "y": 380,
        "wires": []
    },
    {
        "id": "1819b7821da15b51",
        "type": "api-call-service",
        "z": "f3bbc3f8714fe9d3",
        "name": "",
        "server": "c4b4e1462d887a8e",
        "version": 5,
        "debugenabled": false,
        "domain": "light",
        "service": "turn_on",
        "areaId": [],
        "deviceId": [
            "721f370ef251e91b3508fb2dd54a69ed"
        ],
        "entityId": [
            "light.partition_light_1"
        ],
        "data": "{\"rgb_color\": [255,255,255]}",
        "dataType": "jsonata",
        "mergeContext": "",
        "mustacheAltTags": false,
        "outputProperties": [],
        "queue": "none",
        "x": 670,
        "y": 360,
        "wires": [
            [
                "a2d99facf5d519ff"
            ]
        ]
    },
    {
        "id": "267b2911db5d34cd",
        "type": "inject",
        "z": "f3bbc3f8714fe9d3",
        "name": "",
        "props": [
            {
                "p": "payload"
            },
            {
                "p": "topic",
                "vt": "str"
            }
        ],
        "repeat": "",
        "crontab": "",
        "once": false,
        "onceDelay": 0.1,
        "topic": "",
        "payload": "",
        "payloadType": "date",
        "x": 440,
        "y": 340,
        "wires": [
            [
                "1819b7821da15b51"
            ]
        ]
    },
    {
        "id": "c4b4e1462d887a8e",
        "type": "server",
        "name": "Home Assistant",
        "version": 5,
        "addon": false,
        "rejectUnauthorizedCerts": true,
        "ha_boolean": "y|yes|true|on|home|open",
        "connectionDelay": true,
        "cacheJson": true,
        "heartbeat": false,
        "heartbeatInterval": "30",
        "areaSelector": "friendlyName",
        "deviceSelector": "friendlyName",
        "entitySelector": "friendlyName",
        "statusSeparator": ": ",
        "statusYear": "hidden",
        "statusMonth": "short",
        "statusDay": "numeric",
        "statusHourCycle": "default",
        "statusTimeFormat": "h:m",
        "enableGlobalContextStore": false
    }
]

This works:
curl -X POST -H 'Authorization: Bearer <my token>' -H 'Content-Type: application/json' -d '{"entity_id": "light.partition_light_1"}' http://192.168.1.88:8123/api/services/light/turn_on

I'm almost certainly doing something wrong. Can you point me in the right direction please.

...Andy

**Comment by zachowj** [comment, marked as answer]
Remove the device from the call service nice and only use the entity I'd.

**Reply by asamicros** [reply]
Brilliant. Worked like a charm. Thanks.
